### ISSUE TITLE
Ridgeplot shape

### ISSUE THREAD

**Issue by alnf** [dtm2451/dittoSeq#71]
Is there a way to plot histogram instead of density? Sometimes it is needed for raw counts.

**Comment by dtm2451** [comment]
Not currently, but I could potentially add that... seems I could add it fairly easy ([see here](https://cran.r-project.org/web/packages/ggridges/vignettes/introduction.html#themes), and scroll up just a bit).

In the meantime, you could rely on `dittoRidgeJitter()` (or `dittoPlot(..., plots = c("ridgeplot", "jitter")`) to see via the jitter where cells actually fall.

OR converting this code to your data should get you close to what I'll add sometime soon via an `ridgeplot.scaling = "count"` option.  (Note that none of the built in `ridgeplot.*` inputs will work with the below code because this completely bypasses `dittoPlot()`'s ridgeplot code.)
```
library(dittoSeq)

# Loads in the data for this example
example(importDittoBulk, echo = FALSE)

# Giving 'plots = c()' will ready an empty *y-direction* plot, so
# you'll need to manually fix that the `xlab` & `ylab` would be reversed.
dittoPlot(myRNA, "gene1", group.by = "timepoint",
    plots = c(), min = NA, max = NA,
    xlab = "nCounts", ylab = "groups") +
    ggridges::geom_density_ridges2(aes(x = var.data, y = grouping), stat = "binline")
```
![image](https://user-images.githubusercontent.com/38870347/107441881-e746cb00-6aea-11eb-8109-03389cc83b89.png)

**Comment by dtm2451** [comment]
Okay, ignore the code above... I worked it in already =).  I'm not 100% on how I set it up to work yet so it's in a branch, but you can install the `histogram` branch to test!  `BiocManager::install("dtm2451/dittoSeq@histogram")`.

Currently: `dittoPlot(..., plots = "ridgeplot", ridgeplot.shape = "count")` and optional `ridgeplot.bins` controls how many bins.
I'll make a PR that references this issue before I commit any changes that might change the above behavior, but basically thinking of whether I should make "hist" value that goes directly to `plots` instead, or maybe just change change the `ridgeplot.shape = "count"`  to expect "hist" instead.

**Comment by alnf** [comment]
Ok, I've tried this:
`dittoPlot(plc, "XIST", group.by = "clusters", plots = c("ridgeplot", "jitter"), ridgeplot.shape = "count")`

Pasting part of the image:
![Screenshot from 2021-02-10 19-35-43](https://user-images.githubusercontent.com/2435340/107548374-3e29ce80-6bd7-11eb-9baf-53c2d986bd5e.png)

I guess jitter doesn't make sense in case of histogram at all, the warning I got was:
`Ignoring unknown parameters: jittered_points `
When I removed wrong option - no warning.

How to make it bin exactly by integers? Set `ridgeplot.bins` as max value + 1?

**Comment by alnf** [comment]
Better reproducible example:

```
library(dittoSeq)

# Loads in the data for this example
example(importDittoBulk, echo = FALSE)

# Giving 'plots = c()' will ready an empty *y-direction* plot, so
# you'll need to manually fix that the `xlab` & `ylab` would be reversed.
dittoPlot(myRNA, "gene1", group.by = "timepoint",
          plots = c(), min = NA, max = NA,
          xlab = "nCounts", ylab = "groups") +
  ggridges::geom_density_ridges2(aes(x = var.data, y = grouping), stat = "binline")

dittoPlot(myRNA, "gene1", group.by = "timepoint",
         plots = "ridgeplot", ridgeplot.shape = "count",
         xlab = "nCounts", ylab = "groups",
         min = NA, max = NA)
```
For same reason in a second case the styling is different. I'm just not sure it is expected behavior, so reporting just in case.

![image](https://user-images.githubusercontent.com/2435340/107551165-99a98b80-6bda-11eb-9948-e57b68179ae9.png)
![image](https://user-images.githubusercontent.com/2435340/107551487-00c74000-6bdb-11eb-8d70-aa1bba6ff917.png)

**Comment by dtm2451** [comment]
Re jitter: Yea, jitter will nolonger work in the histogram setting. (The jitter for ridgeplots is actually part of the `ggridges::geom_density_ridges` system with dot location calculated by its associated `stat_density_ridges`. When we change that stat to `stat_binline` for making the histogram representation, the associated jitter input is gone. So you get that expected warning. And it's a warning which I left to show 'as is', thinking that it does the job fine, in lieu of making my own warning/error-handling code.)

Re bin size: Hmmm I can expose `binwidth` also / instead of the `bins`.  It's not there.  But it can be done.

Re styling: In the "first" plot, all the styling of ridges is the ggridges defaults. I suppose maybe their default lineweight is 0.5. In the second, now you are getting the dittoPlot stylings.  The lineweight is based on `ridgeplot.lineweight` which defaults to 1, and the extra space at the bottom is removed as I feel that is wasted space.

And re "hist" versus "count"...  Def understand where you are coming from and thanks for your take! I think I'm going to go with "hist", just still have to decide whether that goes to `ridgeplot.shape` vs `plots` with all tweaks then as `hist.bins` & `hist.binwidth` and so on. My philosophy for dittoSeq has been to stay more in the realm of RNAseq jargon, and away from ggplot jargon, as much as possible, so that plotting with dittoSeq doesn't rely on any prior ggplot knowledge =).

**Comment by dtm2451** [comment]
Oh and PS, you don't need the

```
xlab = "nCounts", ylab = "groups",
         min = NA, max = NA
```

anymore.  That was just to get around problematic things for an x-direction plot that the y-direction dittoPlotter-internal would have set.

**Comment by alnf** [comment]
> And re "hist" versus "count"... Def understand where you are coming from and thanks for your take! I think I'm going to go with "hist", just still have to decide whether that goes to `ridgeplot.shape` vs `plots` with all tweaks then as `hist.bins` & `hist.binwidth` and so on. My philosophy for dittoSeq has been to stay more in the realm of RNAseq jargon, and away from ggplot jargon, as much as possible, so that plotting with dittoSeq doesn't rely on any prior ggplot knowledge =).

Making it consistent across the package is also important! So go with anything you prefer, as long as it's in the manual, I'm sure people won't have much troubles using.

Thx for your quick replies, I'm happy now than I can use histogram:)

**Comment by dtm2451** [comment]
Finally able to finalize these things!*

Settled on keeping the `ridgeplot.shape` input, but I changed the value needed for histograms to "hist".  (Also changed the ridgeplot value from density to "smooth", but that's less important as probably no one will ever need to give that!)

I also added a `ridgeplot.binwidth` input.  As in the underlying `?ggrides::stat_bin_width`, precedence goes to this input, rather than `ridgeplot.bins`, if both are given.

Lemme know if this all works for you!

*There actually are multiple fires to put out that I just noticed (I think they're actually new today, but :shrug:?) before I can actually merge it in / get the update into Bioconductor.  So unfortunately, still requires `BiocManager::install("dtm2451/dittoSeq@histogram")`.
